### PR TITLE
[Merged by Bors] - feat(UniformConvergenceTopology): prove generic `UniformOnFun.continuousOn_eval₂`

### PR DESCRIPTION
**Pull request by urkud**
Also use it to weaken TC assumptions in `ContinuousMultilinearMap.continuous_eval`.

---

- [ ] depends on: #13531

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/NormedSpace/Multilinear/Basic.lean [hunk 1]
```diff
@@ -76,6 +76,20 @@ We use the following type variables in this file:
...
 universe u v v' wE wE₁ wE' wG wG'
 
+/-- Applying a multilinear map to a vector is continuous in both coordinates. -/
+theorem ContinuousMultilinearMap.continuous_eval {𝕜 ι : Type*} {E : ι → Type*} {F : Type*}
+    [NormedField 𝕜] [Finite ι] [∀ i, SeminormedAddCommGroup (E i)] [∀ i, NormedSpace 𝕜 (E i)]
+    [TopologicalSpace F] [AddCommGroup F] [TopologicalAddGroup F] [Module 𝕜 F] :
+    Continuous fun p : ContinuousMultilinearMap 𝕜 E F × ∀ i, E i => p.1 p.2 := by
+  cases nonempty_fintype ι
+  letI := TopologicalAddGroup.toUniformSpace F
+  haveI := comm_topologicalAddGroup_is_uniform (G := F)
```
Comment (dupuisf) [inline Mathlib/Analysis/NormedSpace/Multilinear/Basic.lean:86]:
```suggestion
  let := TopologicalAddGroup.toUniformSpace F
  have := comm_topologicalAddGroup_is_uniform (G := F)
```
Are `letI` and `haveI` still useful in Lean 4?

**Review by dupuisf — COMMENTED** [review]
Thanks!

bors d+

File: Mathlib/Analysis/NormedSpace/Multilinear/Basic.lean (hunk 1 above)
Comment (urkud) [reply Mathlib/Analysis/NormedSpace/Multilinear/Basic.lean:86]: AFAIR, I used `letI` to avoid `_`.

Comment (urkud) [reply Mathlib/Analysis/NormedSpace/Multilinear/Basic.lean]: They are useful, if you want to inline your `let`.

**Pull request closed without merging**